controllers/flinkcluster_reconciler.go

```diff
@@ -401,11 +403,9 @@ func (reconciler *ClusterReconciler) reconcileJob() (ctrl.Result, error) {
 
 	// Create
 	if desiredJob != nil && observedJob == nil {
-		// If the observed Flink job status list is not nil (e.g., emtpy list),
-		// it means Flink REST API server is up and running. It is the source of
-		// truth of whether we can submit a job.
-		if observed.flinkJobList == nil {
-			log.Info("Waiting for Flink API server to be ready")
+		// Proceed job creation process when the cluster components are updated and Flink API server is ready
+		if !isClusterReady(observed) {
+			log.Info("Waiting for cluster components to be updated and Flink API server to be ready")
```

Comment:
s/updated/ready, it is not necessarily to be "updated" here.

Reply:
Done.